src/symbol/annotations.rs

```diff
@@ -0,0 +1,290 @@
+    /// param : how many lines, default 1
+    ChangeLineEndDelta = 7,
+    /// param : either 1 (default, for statement)
+    ///         or 0 (for expression)
+    ChangeRangeKind = 8,
+    /// param : start column number, 0 means no column info
+    ChangeColumnStart = 9,
+    /// param : end column number delta (signed)
+    ChangeColumnEndDelta = 10,
+    /// param : ((sourceDelta << 4) | CodeDelta)
+    ChangeCodeOffsetAndLineOffset = 11,
+    /// param : codeLength, codeOffset
+    ChangeCodeLengthAndCodeOffset = 12,
+    /// param : end column number
+    ChangeColumnEnd = 13,
+    /// A non valid value
+    Invalid,
+}
+
+impl From<u32> for BinaryAnnotationOpcode {
+    fn from(value: u32) -> Self {
+        match value {
+            0 => BinaryAnnotationOpcode::Eof,
+            1 => BinaryAnnotationOpcode::CodeOffset,
+            2 => BinaryAnnotationOpcode::ChangeCodeOffsetBase,
+            3 => BinaryAnnotationOpcode::ChangeCodeOffset,
+            4 => BinaryAnnotationOpcode::ChangeCodeLength,
+            5 => BinaryAnnotationOpcode::ChangeFile,
+            6 => BinaryAnnotationOpcode::ChangeLineOffset,
+            7 => BinaryAnnotationOpcode::ChangeLineEndDelta,
+            8 => BinaryAnnotationOpcode::ChangeRangeKind,
+            9 => BinaryAnnotationOpcode::ChangeColumnStart,
+            10 => BinaryAnnotationOpcode::ChangeColumnEndDelta,
+            11 => BinaryAnnotationOpcode::ChangeCodeOffsetAndLineOffset,
+            12 => BinaryAnnotationOpcode::ChangeCodeLengthAndCodeOffset,
+            13 => BinaryAnnotationOpcode::ChangeColumnEnd,
+            _ => BinaryAnnotationOpcode::Invalid,
+        }
+    }
+}
+
+/// Represents a parsed `BinaryAnnotation`.
+///
+/// Binary annotations are used by `S_INLINESITE` to encode opcodes for how to
+/// evaluate the state changes for inline information.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum BinaryAnnotation {
+    CodeOffset(u32),
+    ChangeCodeOffsetBase(u32),
+    ChangeCodeOffset(u32),
+    ChangeCodeLength(u32),
+    ChangeFile(FileIndex),
+    ChangeLineOffset(i32),
+    ChangeLineEndDelta(u32),
+    ChangeRangeKind(u32),
+    ChangeColumnStart(u32),
+    ChangeColumnEndDelta(i32),
+    ChangeCodeOffsetAndLineOffset(u32, i32),
+    ChangeCodeLengthAndCodeOffset(u32, u32),
+    ChangeColumnEnd(u32),
+}
+
+impl BinaryAnnotation {
+    /// Does this annotation emit a line info?
+    pub fn emits_line_info(self) -> bool {
+        match self {
+            BinaryAnnotation::ChangeCodeOffset(..) => true,
+            BinaryAnnotation::ChangeCodeOffsetAndLineOffset(..) => true,
+            BinaryAnnotation::ChangeCodeLengthAndCodeOffset(..) => true,
+            _ => false,
+        }
+    }
+}
+
+/// An iterator over binary annotations used by `S_INLINESITE`.
+#[derive(Clone, Debug, Default)]
+pub struct BinaryAnnotationsIter<'t> {
+    buffer: ParseBuffer<'t>,
+}
+
+impl<'t> BinaryAnnotationsIter<'t> {
+    fn uncompress_next(&mut self) -> Result<u32> {
```

Comment:
https://github.com/willglynn/pdb/pull/51#discussion_r320559874

> Does this variable length encoding scheme have a name? Perhaps equivalently, does this encoding scheme exist anywhere else? It's not LEB128 like DWARF, it's not varint like Protocol Buffers, it's apparently not even anywhere else in the PDB, and I didn't find any match [in the Wikipedia list](https://en.wikipedia.org/wiki/Variable-length_quantity).
>
> If this _does_ have a name, I'd like to use it. If it truly is ad hoc and confined to exactly this scenario, well, `BinaryAnnotationsIter` sounds good 😄

Reply:
This is literally called `CVCompressData` and `CVUncompressData`. I'll copy the doc blocks over for clarity 👍